### PR TITLE
Minor fixes and preparations for adding file lists to `OrtResult`

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Part of #6944.

### CODE REVIEW

File: model/src/main/kotlin/utils/SortedSetConverters.kt [hunk 1]
```diff
@@ -77,7 +77,7 @@ class ScanResultSortedSetConverter : StdConverter<Set<ScanResult>, Set<ScanResul
                     this += it.provenance.sourceArtifact.url
                 }
                 else -> {
-                    // Cannot happen as ScanResults in ScannerRun
+                    // Do not append anything, because unknown provenance does not have any properties.
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/SortedSetConverters.kt]: Nit: "append" is the verb used for `buildString`, but for `buildList` it'd be "add".

File: model/src/main/kotlin/utils/SortedSetConverters.kt [hunk 2]
```diff
@@ -93,4 +77,22 @@ class SnippetFindingSortedSetConverter : StdConverter<Set<SnippetFinding>, Sorte
         value.toSortedSet(compareBy<SnippetFinding> { it.sourceLocation.path }.thenByDescending { it.snippet.purl })
 }
 
+private fun Provenance.sortKey(): String =
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/SortedSetConverters.kt:80]: The "sort" verb as part of the function name sounds as if the function itself would already sort. Maybe name this "toKey()", "getSortKey()", "getSelector()" or similar?

**Review by sschuberth — CHANGES_REQUESTED** [review]
Generally, it's a bit hard to judge whether the refactorings make sense when the announced upcoming usages are not in this PR.

**Comment by fviernau** [comment]
> Generally, it's a bit hard to judge whether the refactorings make sense when the announced upcoming usages are not in this PR.

Do you want me to close this PR?

**Comment by fviernau** [comment]
> Not necessarily, I just wanted to point this out as a reviewer. Here, the refactorings are probably uncritical (though it seems a bit over the top that we have three variants of `prependPath` now), but for bigger code extractions for reuse it'd be nice to know what the reuse is. Because if the review turns out that the reuse does not make sense / is not needed (for whatever reason), the refactoring doesn't either, so it's easier to review both in the same context.

The planned use of the new `prependPath` is: https://github.com/oss-review-toolkit/ort/pull/7164/commits/077b5948a664de0e4f748dd7cfeb7154ec782e59#diff-c530c7a3069096e7ad7b11deffe116249ef3cc3938579dbdf08d9f22e7b030f3R290

**Review by sschuberth — APPROVED** [review]

**Pull request merged**